subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java

```diff
@@ -307,7 +325,7 @@ void setRequiredAttributes(@Nullable Map<Attribute<Object>, Set<Object>> require
     }
 
     private static class ContentSpec {
-        private final boolean regex;
+        private final MatcherKind matcherKind;
         private final String group;
         private final String module;
```

Comment:
❔ Shouldn't there be `@Nullable` on the fields here, too?

Reply:
Maybe? It's not strictly necessary, they're unspecified nullability by default. I'd rather leave them.